### ISSUE TITLE
fastapi function check too strict

### ISSUE THREAD

**Issue by uriva** [tiangolo/fastapi#757]
I want to use fastapi with callables which return False on a `inspect.is_function` check.

```Python
from fastapi import FastAPI
import toolz

def some_function():
    ...

def error_handler():
    ...

endpoint = toolz.excepts(Exception, some_function, error_handler)
app = FastAPI()
app.get("/")(endpoint)

```

Result:

> AssertionError: An endpoint must be a function or method

**Comment by prostomarkeloff** [comment]
Interesting situation. But I want suggest to you a much clear way for exceptions handling. 
When you are trying handle almost **all** exceptions (with `Exception`) you will handle any exceptions and it's wrong way. But if you really want handle any exceptions in your application, you may try [use it](https://fastapi.tiangolo.com/tutorial/handling-errors/). But if you really want do it - try it:
```py
@app.get("/")
async def func("Place here your args and kwargs"):
    return your_callable(*args, **kwargs)
```

**Comment by prostomarkeloff** [comment]
However, i think we should create PR that will add possibility of adding any callable objects.

**Comment by uriva** [comment]
Yeah it's just an example, `functools.partial` and I guess `toolz.compose` don't work either.

**Comment by tiangolo** [comment]
Yeah. This makes sense. I was going to document a working example, etc. But it was faster to fix it :man_shrugging: :tada: 

This should be fixed by #977
